### ISSUE TITLE
Add Menu Filtering for the Word Alignment Tool

### ISSUE THREAD

**Issue by benjore** [unfoldingWord/translationCore#4443]
## Specs

- [x] Verbiage
    - Invalidated
    - Bookmark
    - Completed
    - Incomplete
    - Verse edit
    - Comments
- [x] Same functionality as in the tW tool (see #1998 )
  - [x] If `Completed` is selected, then `Incomplete` becomes inactive
  - [x] If multiple items on the filter are selected, it is OR (not AND). 
  - [x] Filter needs to collapse, show chips, etc. 
  - [x] Add spinner

## Mockups
*(From tW tool)*
<img width="236" alt="screen shot 2018-10-15 at 2 09 40 pm" src="https://user-images.githubusercontent.com/13352518/46972399-11eff880-d084-11e8-85a1-23d7f7a94c31.png">
- [x] Use the same icons as in the tW tool. Note the verbiage changes between `Completed` (in wA tool) and `Selected` (in tW tool)

**Comment by mannycolon** [comment]
- Make the menu filtering functionality dynamic so that it handles multiple tools.
- Basically, generate the filter options base on the tools checking data.

**Comment by da1nerd** [comment]
are the comments and bookmarks shared between tools or are they separate?
For now the are separate.

**Comment by da1nerd** [comment]
Discovered this bug waiting to happen while investigating the menu update. https://github.com/unfoldingWord-dev/translationCore/issues/5592

**Comment by da1nerd** [comment]
Here is a proposal to make the menu more modular and easier for tools to manage https://codesandbox.io/s/xvq9zx3pyw

**Comment by da1nerd** [comment]
It took some digging but I finally discovered that the group menu items are receiving their filter booleans within `js/actions/GroupsDataActions.js` in the `verifyGroupDataMatchesWithFs` method.

It will be complicated to add support for wA filtering here because wA is managing a lot of it's own state. There are a number of different filtering keys with differentiating logic that will need to be added to wA. If we follow the same methodology as in tC we'll need add in a lot of custom reducer and action logic for wA apart from what already exists for tW. This will only increase in complexity if we add more filters and especially if we add more tools.

## Proposal
A more maintainable approach would be the following:
1. move all checking data state out of tC and into the tools.
2. Create api methods on all of the tools for accessing the different checking state.
3. update tC to call the appropriate api methods to retrieve checking state and set booleans on the group data items.

This has several benefits:

* decouples checking state from tC for better tool abstraction.
* greatly reduces the work required to support additional filters/tools.
* allows tools to manage their own checking state however they like without conflicting with other tools.
* combined with the proposed structure in https://codesandbox.io/s/xvq9zx3pyw we could even automate the entire filter. So that any check data provided by a tool could be automatically supported by the filter.

## First step

The above would require a multi-stage refactor. Therefore, in the mean time I could simply manually insert the booleans into the group data items within wA, which would effectively bypass the entire system in tC. That would require the minimal amount of time and would not require any refactoring. It would also not add to the complexity of a future refactor should we choose to follow through with the above proposal.

I believe this small first step would be faster than hooking up wA filtering into the existing process in tC.

**Comment by da1nerd** [comment]
@benjore thoughts on the above :point_up:

It may seem ironic but I promise I'm not purposely trying to refactor all-the-things. :smile:

**Comment by benjore** [comment]
I'm neither for or against. My concern is that it works properly when it's done. How you do it is between you and the rest of the team. I'm all for sustainable code.

**Comment by da1nerd** [comment]
Last few items remaining:

- [ ] performance when used in tW (rendering is slow)
- [x] use correct font
- [x] abstract the active menu logic item behind an `itemId` field so that matching criteria can be specified in the tool. 
- [ ] [clear checkmark after verse is edited in tW](https://github.com/unfoldingWord-dev/translationCore/issues/5654) may include this in a separate PR.

**Comment by da1nerd** [comment]
I turned this into an Epic because there are 4 different PRs related to it.

**Comment by cckozie** [comment]
Waiting on #5710 since it has problems that relate to both tW and wA.

**Comment by cckozie** [comment]
1.1.3 (e1fae24) 
There is no spinner, but I do not believe that it is necessary and Ben agrees.
I'm assuming that when #5832 has been implemented the verses edited in tW (or tN) will show when the verse edit filter is on.
